File: Leetcode 150/Array_String/13-roman-to-integer/roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        number = 0
        roman = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        for i,j in zip(s,s[1:]):
            if roman[i]<roman[j]:
                number -= roman[i]
            else:
                number += roman[i]
        return number + roman[s[len(s)-1]]
        

File: Leetcode 150/Array_String/13-roman-to-integer/roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        int number =0;
        Map<Character, Integer> hash = new HashMap<>();
        hash.put('I', 1);
        hash.put('V', 5);
        hash.put('X', 10);
        hash.put('L', 50);
        hash.put('C', 100);
        hash.put('D', 500);
        hash.put('M', 1000);

        for(int i=0;i<s.length()-1;i++)
        {
            if(hash.get(s.charAt(i))<hash.get(s.charAt(i+1)))
            {
                number-=hash.get(s.charAt(i));
            }
            else
            {
                number+=hash.get(s.charAt(i));
            }
        }
        return number+hash.get(s.charAt(s.length()-1));
    }
}